include/boost/multiprecision/detail/default_ops.hpp

```diff
@@ -2616,14 +2619,29 @@ inline BOOST_MP_CXX14_CONSTEXPR typename multiprecision::detail::expression<tag,
 template <class Backend, multiprecision::expression_template_option ExpressionTemplates>
 inline BOOST_MP_CXX14_CONSTEXPR multiprecision::number<Backend, ExpressionTemplates> erf BOOST_PREVENT_MACRO_SUBSTITUTION(const multiprecision::number<Backend, ExpressionTemplates>& arg)
 {
+   const int fpc_arg { (boost::multiprecision::fpclassify)(arg) };
+
+   if (fpc_arg == FP_ZERO)
+   {
+      return arg;
+   }
+
    detail::scoped_default_precision<multiprecision::number<Backend, ExpressionTemplates> > precision_guard(arg);
    return boost::math::erf(arg, c99_error_policy());
 }
 template <class tag, class A1, class A2, class A3, class A4>
 inline BOOST_MP_CXX14_CONSTEXPR typename multiprecision::detail::expression<tag, A1, A2, A3, A4>::result_type erf BOOST_PREVENT_MACRO_SUBSTITUTION(const multiprecision::detail::expression<tag, A1, A2, A3, A4>& arg)
 {
    using value_type = typename multiprecision::detail::expression<tag, A1, A2, A3, A4>::result_type;
-   detail::scoped_default_precision<value_type>                                          precision_guard(arg);
+   detail::scoped_default_precision<value_type> precision_guard(arg);
+
+   const int fpc_arg { (boost::multiprecision::fpclassify)(arg) };
```

Comment:
As above, but this is even worse, since we evaluate the same expression template *twice* here, once for fpclassify and once for the erf call.

Reply:
I'm working out a sensible repair in Math, partly based on your suggestion below. Thanks @jzmaddock